polkadot/node/network/collator-protocol/src/collator_side/metrics.rs

```diff
@@ -116,8 +165,323 @@ impl metrics::Metrics for Metrics {
 				)?,
 				registry,
 			)?,
+			collation_fetch_latency: prometheus::register(
+				prometheus::Histogram::with_opts(
+					prometheus::HistogramOpts::new(
+						"polkadot_parachain_collation_fetch_latency",
+						"How much time collations spend waiting to be fetched",
+					)
+					.buckets(vec![
+						0.001, 0.01, 0.025, 0.05, 0.1, 0.15, 0.25, 0.35, 0.5, 0.75, 1.0, 2.0, 5.0,
+					]),
+				)?,
+				registry,
+			)?,
+			collation_backing_latency_time: prometheus::register(
+				prometheus::Histogram::with_opts(
+					prometheus::HistogramOpts::new(
+						"polkadot_parachain_collation_backing_latency_time",
+						"How much time it takes for a fetched collation to be backed",
+					)
+					.buckets(vec![
+						1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 12.0, 15.0, 18.0, 24.0, 30.0,
+					]),
+				)?,
+				registry,
+			)?,
+			collation_backing_latency: prometheus::register(
+				prometheus::Histogram::with_opts(
+					prometheus::HistogramOpts::new(
+						"polkadot_parachain_collation_backing_latency",
+						"How many blocks away from the relay parent are collations backed",
+					)
+					.buckets(vec![1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0]),
+				)?,
+				registry,
+			)?,
+			collation_inclusion_latency: prometheus::register(
+				prometheus::Histogram::with_opts(
+					prometheus::HistogramOpts::new(
+						"polkadot_parachain_collation_inclusion_latency",
+						"How many blocks it takes for a backed collation to be included",
+					)
+					.buckets(vec![1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]),
+				)?,
+				registry,
+			)?,
+			collation_expired_total: prometheus::register(
+				prometheus::HistogramVec::new(
+					prometheus::HistogramOpts::new(
+						"polkadot_parachain_collation_expired",
+						"How many collations expired (not backed or not included)",
+					)
+					.buckets(vec![1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0]),
+					&["state"],
+				)?,
+				registry,
+			)?,
 		};
 
 		Ok(Metrics(Some(metrics)))
 	}
 }
+
+// Equal to allowed ancestry len + 1
```

Comment:
so claim queue len? allowed ancestry len doesn't matter anymore

Reply:
yeah, I was time travelling in the past when I wrote the code :)